src/main/java/edu/hawaii/its/groupings/service/OotbActiveUserProfileService.java

```diff
@@ -1,51 +1,64 @@
+    private static final Log logger = LogFactory.getLog(OotbActiveUserProfileService.class);
+    private final Map<String, User> users = new LinkedHashMap<>();
 
     public OotbActiveUserProfileService() {
         initUsers();
     }
 
     private void initUsers() {
-        // Initialize member with uid "member0123"
-        users.put("MEMBER", new User.Builder("member0123")  // UID is clearly the first parameter in the constructor
-                .uhUuid("11111111")
-                .authorities(AuthorityUtils.createAuthorityList("ROLE_UH", "ROLE_OOTB"))
-                .addAttribute("cn", "MEMBER")
-                .addAttribute("mail", "member@hawaii.edu")
-                .addAttribute("givenName", "DefaultMember")
-                .build());
-
-        // Initialize owner with uid "owner0123"
-        users.put("OWNER", new User.Builder("owner0123")
-                .uhUuid("22222222")
-                .authorities(AuthorityUtils.createAuthorityList("ROLE_UH", "ROLE_OWNER", "ROLE_OOTB"))
-                .addAttribute("cn", "OWNER")
-                .addAttribute("mail", "owner@hawaii.edu")
-                .addAttribute("givenName", "OwnerUser")
-                .build());
-
-        // Initialize admin with uid "admin0123"
-        users.put("ADMIN", new User.Builder("admin0123")
-                .uhUuid("33333333")
-                .authorities(AuthorityUtils.createAuthorityList("ROLE_ADMIN", "ROLE_UH", "ROLE_OWNER", "ROLE_OOTB"))
-                .addAttribute("cn", "ADMIN")
-                .addAttribute("mail", "admin@hawaii.edu")
-                .addAttribute("givenName", "AdminUser")
-                .build());
+        try {
+            mapFromJsonFile("ootb.active.user.profiles.json").forEach((key, userData) -> {
+                users.put(key, createUserFromMap(userData));
+            });
+        } catch (IOException e) {
+            logger.error("Error while initiating active user profiles from a json file: ", e);
+        }
+    }
+
+    private Map<String, Map<String, Object>> mapFromJsonFile(String filepath) throws IOException {
+        File file = new ClassPathResource(filepath).getFile();
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        // Map<ootb active profile type, Map<profile property, property's value>>
+        return objectMapper.readValue(file, new TypeReference<>() {});
+    }
+
+    private User createUserFromMap(Map<String, Object> userData) {
+        String uid = (String) userData.get("uid");
+        String uhUuid = (String) userData.get("uhUuid");
+        List<String> roles = (List<String>) userData.get("authorities");
+        Map<String, String> attributes = (Map<String, String>) userData.get("attributes");
+
+        User.Builder builder = new User.Builder(uid)
+                .uhUuid(uhUuid)
+                .addAuthorities(roles);
+
+        attributes.forEach(builder::addAttribute);
+
+        return builder.build();
```

Comment:
I will say I do prefer your old code before the refactor because it was a lot simpler than the code that reads JSON files. You can make the call whether you think this refactor is needed or not.

I think your goal with this ticket was to put the default users inside a properties or JSON file. I would have gone with how you load data from the `data.harness` on the API side and put the JSON objects inside the application-ootb.properties or add a data.harness file to the UI side. Then use the JsonUtil class to load it. That way we can reuse the code for loading JSON files instead of adding new code here.

Reply:
Looks nice so far... but maybe using the Spring "@Resource" annotation would be good here. e.g., @Resource("${ootb.active.user.profiles.json}").  

The casts after reading the file are bit iffy. I think we have a helper method in the JsonUtil class to create objects. I think we should use that to create the User objects. If the helper method doesn't do exactly what you need, maybe you can create another helper method in JsonUtil.

Reply:
I also actually prefer the code before the refactor. However, during the last meeting, it was mentioned that the ootb project should serve as a repository with a variety of purposes beyond just showing the groupings project without the grouper client connection. The goal is for developers to be able to create various scenarios, use and test them directly in the development of groupings. During that discussion, it was mentioned that using JSON would make it easier for developers to view and modify the data, and easily import it with an overrides file. Additionally, when we later create something like a QA tool, it would be synergistic to have it in JSON format. Initially, I considered loading the JSON file into the properties file like the existing data harness method and calling it with propertyLocator, but I thought it would be difficult for developers to modify the data later if we compact the JSON data into a single line. So instead of properties, I decided to create it as a separate JSON file.

Reply:
> I will say I do prefer your old code before the refactor because it was a lot simpler than the code that reads JSON files. You can make the call whether you think this refactor is needed or not.
> 
> I think your goal with this ticket was to put the default users inside a properties or JSON file. I would have gone with how you load data from the `data.harness` on the API side and put the JSON objects inside the `application-ootb.properties` or add a `data.harness` file to the UI side. Then use the JsonUtil class to load it. That way we can reuse the code for loading JSON files instead of adding new code here.

I also actually prefer the code before the refactor. However, during the last meeting, it was mentioned that the ootb project should serve as a repository with a variety of purposes beyond just showing the groupings project without the grouper client connection. The goal is for developers to be able to create various scenarios, use and test them directly in the development of groupings. During that discussion, it was mentioned that using JSON would make it easier for developers to view and modify the data, and easily import it with an overrides file. Additionally, when we later create something like a QA tool, it would be synergistic to have it in JSON format. Initially, I considered loading the JSON file into the properties file like the existing data harness method and calling it with propertyLocator, but I thought it would be difficult for developers to modify the data later if we compact the JSON data into a single line. So instead of properties, I decided to create it as a separate JSON file.